cmd/conf/chain.go

```diff
@@ -52,27 +56,44 @@ func (c *ParentChainConfig) Validate() error {
 	return c.Connection.Validate()
 }
 
+// lint:require-exhaustive-initialization
 type L2Config struct {
-	ID        uint64                   `koanf:"id"`
-	Name      string                   `koanf:"name"`
-	InfoFiles []string                 `koanf:"info-files"`
-	InfoJson  string                   `koanf:"info-json"`
-	DevWallet genericconf.WalletConfig `koanf:"dev-wallet"`
+	ID               uint64                   `koanf:"id"`
+	Name             string                   `koanf:"name"`
+	InfoFiles        []string                 `koanf:"info-files"`
+	InfoJson         string                   `koanf:"info-json"`
+	InitialL1BaseFee string                   `koanf:"initial-l1base-fee"`
+	DevWallet        genericconf.WalletConfig `koanf:"dev-wallet"`
+}
+
+func (c *L2Config) InitialL1BaseFeeParsed() *big.Int {
+	if c.InitialL1BaseFee == "" {
+		return params.DefaultInitialL1BaseFee
+	}
+
+	parsed, success := big.NewInt(0).SetString(c.InitialL1BaseFee, 10)
+	if !success {
+		log.Error("Failed to parse L1 BaseFee for L2 config", "config", c.InitialL1BaseFee)
```

Comment:
nitpick: Using the flag path, or just the `initial-l1base-fee`, in this error log can help users identify where the problem is.

Reply:
This pattern of falling back to a default if the user provided an invalid argument can create confusion down the line.
WDYT of returning an error here?

Example: User provides 10o instead of 100.
Later this will cause a consensuParsedInitMessage/executionParsedInitMessage mismatch, due to the executionParsedInitMessage.initialL1BaseFee being 50 (default) instead of 100.
The user will need to read the log lines, and likely the code, in order to identify that nitro is ignoring the CLI flag because it is an invalid value.

Reply:
good point!; I included the flag name in the error message and started returning an error in [4947189](https://github.com/OffchainLabs/nitro/pull/4171/commits/4947189f8bf7e49a20c32dfdf322f8df7e8c7f7d)